src/software/simulation/physics/physics_ball.cpp

```diff
@@ -1,21 +1,20 @@
 #include "software/simulation/physics/physics_ball.h"
 
-#include <cmath>
-
 #include "shared/constants.h"
 #include "software/simulation/physics/box2d_util.h"
 #include "software/simulation/physics/physics_object_user_data.h"
 
-PhysicsBall::PhysicsBall(std::shared_ptr<b2World> world, const Ball &ball, double mass_kg,
-                         const double gravity)
+PhysicsBall::PhysicsBall(std::shared_ptr<b2World> world, const BallState &ball_state,
+                         const double mass_kg, const double gravity)
```

Comment:
nit: `const` for primitive types isn't necessary

Reply:
`const` still prevents the value from being modified in the function though. It's not a `const &`, which you're right would be unnecessary